### PR TITLE
`y=None` causes trouble in many cases

### PR DESCRIPTION
**Pull request by ottonemo**
in classifier `fit(X, y=None)` causes problems:
-  for inputs with shape `(n,m)` as it will create `y` with shape `(n,1)` whereas shape `(n,)` would be needed
- for inputs with shape `(n,)` as it will try to access the length of each element to create appropriate dummy values (but the child is a number, not a sequence)

I added tests and a fix for these cases, although regression is still missing.

### CODE REVIEW

**Comment by ottonemo** [comment]
I think you can review now @benjamin-work.

File: inferno/net.py [hunk 1]
```diff
@@ -785,6 +785,17 @@ def __init__(
...
         )
 
+    def check_data(self, _, y):
+        if y is None and self.iterator_train is DataLoader:
+            raise ValueError("No y-values are given (y=None). You must "
+                             "implement your own DataLoader for training "
+                             "(and your validation) and supply it using the "
+                             "`iterator_train` and `iterator_valid` "
+                             "parameters respectively.")
+        elif y is None:
```
Comment (benjamin-work) [inline inferno/net.py]: Why do we need this clause?

File: inferno/tests/test_dataset.py [hunk 2]
```diff
@@ -49,6 +50,175 @@ def test_inconsistent_lengths(self, get_len, data):
...
+    ]
+
+    @pytest.fixture
+    def net_cls_1d_clf(self):
+        class MyModule(nn.Module):
+            def __init__(self):
+                super(MyModule, self).__init__()
+                self.dense = nn.Linear(1,1)
+            def forward(self, X):
+                return F.softmax(self.dense(X.float()))
```
Comment (benjamin-work) [inline inferno/tests/test_dataset.py]: Maybe it's not realistic, but for these tests to run, we don't need the final nonlinearity. If you drop it, clf and reg would use the same module and you could safe some almost duplicate code.

File: inferno/net.py (hunk 1 above)
Comment (ottonemo) [inline inferno/net.py]: To ignore the following check that access `y`.

**Comment by ottonemo** [comment]
@benjamin-work please review

**Comment by benjamin-work** [comment]
:+1:

**Pull request merged**